process_sarif/build_results_archive.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022 Open Source Robotics Foundation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import os
+import os.path
+import pathlib
+import tarfile
+import shutil
+import sys
+import subprocess
+
+from .sarif import SarifFile
+from .sarif_helpers import remove_duplicate_results, get_sarif_in_build
+from typing import List
+
+
+def main():
+    timestr = timestamp()
+    if not os.path.isdir('log/build_results_archives'):
+        os.makedirs('log/build_results_archives')
+    archive_name = archive_filename(timestr)
+    archive_path = os.path.join('log', 'build_results_archives', archive_name)
+    with tarfile.open(archive_path, 'w:bz2') as archive:
+        with open('timestamp', 'w') as tsfile:
+            tsfile.write(timestr)
+        archive.add('timestamp')
+        with open('colcon-build-cmd', 'w') as build_cmd_file:
+            build_cmd = get_build_cmd()
+            build_cmd_file.write(build_cmd)
+        archive.add('colcon-build-cmd')
+        with open('colcon-test-cmd', 'w') as test_cmd_file:
+            test_cmd = get_test_cmd()
+            test_cmd_file.write(test_cmd)
+        archive.add('colcon-test-cmd')
+        with open('processed-sarif-cmd', 'w') as sarif_cmd_file:
+            sarif_cmd = processed_sarif_cmd()
+            sarif_cmd_file.write(sarif_cmd)
+        archive.add('processed-sarif-cmd')
+
+        sarif_files = get_sarif_in_build(verbose=False)
+
+        remove_duplicate_results(sarif_files)
```

Comment:
Does this work? My intent with this function is that it doesn't modify its inputs and instead returns `List[SarifFile]` that have duplicated removed, but its possible that [the function](https://github.com/space-ros/process_sarif/blob/main/process_sarif/sarif_helpers.py#L88) does in fact modify the argument in-place.

Reply:
Good catch! This is left over from one of my earlier drafts fixing up the remove_duplicates function which did modify in place. ebdca25 saves the processed files and writes _them_ to the processed subdir instead of the original SARIF files.